sqlmesh/core/context.py

```diff
@@ -1846,15 +1848,77 @@ def create_external_models(self, strict: bool = False) -> None:
                 strict=strict,
             )
 
+    def get_sensitive_fields(self) -> Set[str]:
+        """Returns a set of common sensitive field names"""
+        return {
+            "password",
+            "secret",
+            "key",
+            "token",
+            "api_key",
+            "private_key",
+            "access_token",
+            "refresh_token",
+            "auth_token",
+            "credentials",
+            "client_secret",
+            "certificate",
+            "ssh_key",
+        }
+
+    def is_sensitive_field(self, field_name: str, sensitive_fields: Set[str]) -> bool:
+        """
+        Check if a field name contains any sensitive keywords
+        """
+        field_lower = field_name.lower()
+        return any(sensitive in field_lower for sensitive in sensitive_fields)
+
+    def mask_sensitive_value(self, value: Any) -> str:
+        """
+        Mask sensitive values with a placeholder
+        Returns '****' for non-empty values and '' for empty ones
+        """
+        if value and str(value).strip():
+            return "*" * len(str(value))
+        return "None"
+
+    def print_config(self, config: ConnectionConfig | None, console: Any) -> None:
+        """
+        Print configuration while masking sensitive information
+
+        Args:
+            config: Pydantic model containing configuration
+            console: Console object with log_status_update method
+        """
+        if not config:
+            console.log_status_update("No connection configuration found.")
+            return
+        sensitive_fields = self.get_sensitive_fields()
+
+        for field_name, value in config.dict().items():
+            if self.is_sensitive_field(field_name, sensitive_fields):
+                masked_value = self.mask_sensitive_value(value)
+                console.log_status_update(f"{field_name}: {masked_value}")
+            else:
+                console.log_status_update(f"{field_name}: {value}")
+
     @python_api_analytics
     def print_info(self, skip_connection: bool = False) -> None:
         """Prints information about connections, models, macros, etc. to the console."""
         self.console.log_status_update(f"Models: {len(self.models)}")
         self.console.log_status_update(f"Macros: {len(self._macros) - len(macro.get_registry())}")
 
+        self.console.log_status_update("\n\nConnection:")
```

Comment:
Should we make the headers more pronounced? Otherwise they blend in with individual connection attributes.

Reply:
How would you do that?

Reply:
You can make it bold by adding appropriate tags, eg:
```
self.console.log_status_update("\n\n[bold]Connection:[/bold]")
```
Ideally we also want to indent everything that follows under the configuration. This way we may not even need to do the highlighting. 

Actually, can we just put this into dict and serialize as YAML instead? Eg.:
```
config = {"Test Connection": { ... }}
```

Reply:
Ok I can try that